model-customization/data-processing/seed-data-for-knowledge/seed-data-for-knowledge.ipynb

```diff
@@ -0,0 +1,377 @@
+    "        review_seed_examples_file(qna_path, min_seed_examples=5, num_qa_pairs=3)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cd24be75-085f-4949-96f6-a8e15f081e72",
+   "metadata": {},
+   "source": [
+    "## Create Seed Dataset for SDG\n",
+    "\n",
+    "This step creates the seed data for SDG. This data is a JSON filed that contains a combination of the `seed_examples` in the qna.yaml and the chunks from the source document. \n",
+    "\n",
+    "Intermediate seed data files are created for each contribution with the contribution's name included in the file name. For example in the `nfl` contribution, a file containing seed data called `seed_data-nfl.jsonl` would be created in `$WORKSPACE_DIR/nfl`. This file contains a combination of all of the chunks from the NFL source documents and the seed examples in the `qna.yaml` in `$WORKSPACE_DIR/nfl/authoring`.\n",
+    "\n",
+    "After seed data files are created for each contribution, a final `seed_data.jsonl` is created in `$WORKSPACE_DIR`. This file is a concatenation of all of the intermediate `seed_data-{contribution name}.jsonl` files and should be used as an input to SDG."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0b588ec4-e929-43ff-91ce-79a04352b728",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install -qq datasets transformers"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5e12c859-e4fb-4185-a65a-33347d773d24",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from utils.create_seed_dataset import get_seed_dataset, safe_concatenate_datasets\n",
+    "\n",
+    "contribution_datasets = []\n",
+    "for contribution in contributions:\n",
+    "    chunks_dir = contribution[\"dir\"]\n",
+    "    qna_dir = contribution[\"dir\"]\n",
+    "    seed_data = get_seed_dataset(chunks_dir, qna_dir)\n",
+    "    output_path = f'{contribution_dir}/seed_data-{contribution_name}.jsonl'\n",
+    "    seed_data.to_json(output_path, orient='records', lines=True)\n",
+    "    contribution_datasets.append(seed_data)\n",
+    "    print(f\"Intermediate results saved to: {output_path}\")\n",
+    "\n",
+    "final_seed_data = safe_concatenate_datasets(contribution_datasets)\n",
+    "output_path = f'{WORKSPACE_DIR}/seed_data.jsonl'\n",
+    "final_seed_data.to_json(output_path, orient='records', lines=True)\n",
+    "\n",
+    "print(f\"Final seed data contains {final_seed_data.data.num_rows} rows\")\n",
+    "print(f\"Final seed data for SDG saved to: {output_path}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a55a0935-11fc-48a4-8866-f9376ebb01fb",
+   "metadata": {},
+   "source": [
+    "### Inspect the seed data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ce1aa362-221a-4a40-b49b-b729194ebf59",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(seed_data.data.table.slice(length=1))"
```

Comment:
Can we do better than just printing the first one? (This can also be a followup PR.)